src/support_sphere/lib/presentation/components/auth/signup_form.dart

```diff
@@ -48,22 +47,50 @@ class SignupForm extends StatelessWidget {
   }
 }
 
+/// Function to validate the form fields
+///
+/// Takes in a list of [validators], the input [value],
+/// and build [context] containing [SignupCubit] as arguments.
+/// It will return the error message if the value is invalid
+/// and null if the value is valid.
+/// Also, it will set the [SignupState.isValid] flag based on
+/// the validity of the value.
+String? validateValue(List<FormFieldValidator<String?>> validators,
+    String? value, BuildContext context) {
+  Function validate = FormBuilderValidators.compose(validators);
+  String? validateResult = validate(value);
+  if (validateResult != null) {
+    context.read<SignupCubit>().setInvalid();
+    return validateResult;
+  }
+  context.read<SignupCubit>().setValid();
+  return null;
```

Comment:
I feel like this function can be extracted out and have the `Cubit` as input as well.. since currently it repeats with login. But that can be done in separate PRs